### PR TITLE
Update installation

### PR DESCRIPTION
**Pull request by Huanli-Meng**
### Motivation

There are some inconsistent descriptions in the Installation Guide. Therefore, fix this inconsistency and refactor the doc structure for Function Mesh.

### Motivation
- Update the Installation guide
- Add a reference section and put the Function Mesh operator configs under this section
- Update the `sidebar.json` file.

### CODE REVIEW

File: docs/function-mesh/function-mesh-crd.md [hunk 1]
```diff
@@ -23,66 +23,66 @@ metadata:
...
+        - persistent://public/default/input
+        typeClassName: "[B"
+      sinkConfig:
+        elasticSearchUrl: "http://quickstart-es-http.default.svc.cluster.local:9200"
+        indexName: "my_index"
+        typeName: "doc"
+        username: "elastic"
+        password: "wJ757TmoXEd941kXm07Z2GW3"
+      pulsar:
+        pulsarConfig: "test-sink"
```
Comment (tpiperatgod) [inline docs/function-mesh/function-mesh-crd.md:42]: this should be `mesh-test-pulsar`, the same as the following Function

File: docs/function-mesh/run-function-mesh.md [hunk 2]
```diff
@@ -29,66 +29,66 @@ To create a Function Mesh, follow these steps.
...
+            - persistent://public/default/input
+            typeClassName: "[B"
+          sinkConfig:
+            elasticSearchUrl: "http://quickstart-es-http.default.svc.cluster.local:9200"
+            indexName: "my_index"
+            typeName: "doc"
+            username: "elastic"
+            password: "wJ757TmoXEd941kXm07Z2GW3"
+          pulsar:
+            pulsarConfig: "test-sink"
```
Comment (tpiperatgod) [inline docs/function-mesh/run-function-mesh.md:48]: ibid

**Review by teachwrite — APPROVED** [review]
LGTM

**Comment by Huanli-Meng** [comment]
@tpiperatgod Thanks for your valuable comments. Docs have been updated. PTAL again, and feel free to approve it if no more comments.

**Review by tpiperatgod — APPROVED** [review]

**Pull request merged**